golem-worker-service/src/api/openapi/converter.rs

```diff
@@ -0,0 +1,1011 @@
+                    for response in map.values_mut() {
+                        response.headers = Some(Self::create_cors_headers("*"));
+                    }
+                    map
+                },
+                security: None,
+                tags: Some(vec![route.template_name.clone()]),
+            },
+            BindingType::Worker => Operation {
+                summary: Some(route.description.clone()),
+                description: None,
+                operation_id: Some(format!("{}_{}",
+                    route.template_name.to_snake_case(),
+                    route.method.to_string().to_lowercase())),
+                parameters: {
+                    let mut params = Self::extract_path_parameters(&route.path).unwrap_or_default();
+                    params.extend(Self::extract_query_parameters(route));
+                    params.extend(Self::extract_header_parameters(route));
+                    if params.is_empty() { None } else { Some(params) }
+                },
+                request_body: Self::create_request_body(route),
+                responses: {
+                    let mut map = Self::create_responses(route, "*");
+                    // Add CORS headers to all responses
+                    for response in map.values_mut() {
+                        response.headers = Some(Self::create_cors_headers("*"));
+                    }
+                    map
+                },
+                security: None,
+                tags: Some(vec![route.template_name.clone()]),
+            },
+            BindingType::Proxy => Operation {
+                summary: Some(route.description.clone()),
+                description: None,
+                operation_id: Some(format!("{}_{}",
+                    route.template_name.to_snake_case(),
+                    route.method.to_string().to_lowercase())),
+                parameters: {
+                    let mut params = Self::extract_path_parameters(&route.path).unwrap_or_default();
+                    params.extend(Self::extract_query_parameters(route));
+                    params.extend(Self::extract_header_parameters(route));
+                    if params.is_empty() { None } else { Some(params) }
+                },
+                request_body: Self::create_request_body(route),
+                responses: {
+                    let mut map = Self::create_responses(route, "*");
+                    // Add CORS headers to all responses
+                    for response in map.values_mut() {
+                        response.headers = Some(Self::create_cors_headers("*"));
+                    }
+                    map
+                },
+                security: None,
+                tags: Some(vec![route.template_name.clone()]),
+            },
+        }
+    }
+
+    fn extract_path_parameters(path: &str) -> Option<Vec<Parameter>> {
+        let params: Vec<Parameter> = path
+            .split('/')
```

Comment:
It is a bad idea to parse this yourself, because it can diverge from the official parser in the worker service (which already supports extensions such as `{foo..}` for multi-segment matches; and will eventually support all RFC-driven extensions).

Reply:
You are right then, I am gonna use `AllPathPatterns::parse()` to properly parse paths then extract path parameters from the parsed patterns. And map the path parameters to OpenAPI Parameter objects.